File: apps/api/src/routes/profiles.ts
import { Hono } from 'hono'
import { prisma } from '@qtscout/db'
import { requireAuth } from '../middleware/auth'
import type { AppEnv } from '../types'

export const profiles = new Hono<AppEnv>()

profiles.get('/leaders', requireAuth, async (c) => {
  const profiles = await prisma.profile.findMany({
    where: { role: { in: ['ADMIN', 'LEADER'] } },
    select: { id: true, name: true, email: true, section: true, roles: true },
    orderBy: { name: 'asc' },
  })
  return c.json({ profiles })
})

File: apps/api/app/routers/profiles.py
from typing import Annotated

from fastapi import APIRouter, Depends
from sqlalchemy import select
from sqlalchemy.ext.asyncio import AsyncSession

from app.db import get_session
from app.deps import CurrentUser
from app.models import Profile
from app.models.enums import UserRole
from app.schemas.user import LeaderProfileOut

router = APIRouter(prefix="/profiles", tags=["profiles"])


@router.get("/leaders")
async def list_leaders(
    user: CurrentUser, session: Annotated[AsyncSession, Depends(get_session)]
) -> dict[str, list[LeaderProfileOut]]:
    result = await session.scalars(
        select(Profile)
        .where(Profile.role.in_([UserRole.ADMIN, UserRole.LEADER]))
        .order_by(Profile.name.asc())
    )
    return {"profiles": [LeaderProfileOut.model_validate(p) for p in result.all()]}
